### ISSUE TITLE
Privacy.sanitize.sanitizeOnShutdown mozilla bug ?

### ISSUE THREAD

**Issue by intika** [intika/Librefox-Firefox#7]
in privafox "privacy.sanitize.sanitizeOnShutdown" is defaulted to true... 

when the config file is used normally with `pref("privacy.sanitize.sanitizeOnShutdown", true); ` the setting is not applied 

when used with mozilla.cfg the setting is enforced BUT its value is string type even if it's set true without quotes... (but the setting work even if its a wrong var)

to be able to default this settings the used value is string so `pref("privacy.sanitize.sanitizeOnShutdown", "true"); ` it works but the side effect is this option can not be disabled in the ffox settings anymore and the gui show a wrong value. 

in short this seems to be a ffox bug where this value is incorrectly set to string instead of bool.

This could affect ghacks too where the setting is just ignored... when set without quotes which is the case.

Any way temporary solution is set with quotes

**Comment by intika** [comment]
solved with defaultPref

**Comment by Thorin-Oakenpants** [comment]
> This could affect ghacks too where the setting is just ignored

Nope. It works as advertised from a user.js. I just tested in a nilla FF65. I did change to custom settings in Options>Privacy & Security>HIstory>Firefo will use<custom settings> so I could see the "clear history when firefox closes" was unchecked.

Added the following user.js

```js
/* TEST ***/
user_pref("_user.js.parrot", "START: Oh yes, the Norwegian Blue... what's wrong with it?");

/* 2802: enable Firefox to clear history items on shutdown
 * [SETTING] Privacy & Security>History>Custom Settings>Clear history when Firefox closes ***/
user_pref("privacy.sanitize.sanitizeOnShutdown", true);

/* END: internal custom pref to test for syntax errors ***/
user_pref("_user.js.parrot", "SUCCESS: No no he's not dead, he's, he's restin'!");
```

Closed and reopened FF, and the value true was applied in about:config (and in prefs.js), and the parrot worked as expected. In options, the setting "Clear history when Firefox closes" was now checked.

The issue must lie with cfg. It's not the first pref I've heard of that has issues there, but not in user.js

**Comment by intika** [comment]
Yes i did checked also in ghack script and found it working it's why i did not report it back... 
Indeed i encountered about 4 or 5 similar cases while enforcing/defaulting settings in the cfg file... it's one of the reason v2 is delayed... but i figure it our finally :)

**Comment by Thorin-Oakenpants** [comment]
Do you mind sharing those troublesome prefs and what you ending up using? As you can see I started a sticky so we can keep track of these sorts of things

**Comment by intika** [comment]
Done ;) https://github.com/ghacksuserjs/ghacks-user.js/issues/601
